key-manager-oapi.yaml

```diff
@@ -0,0 +1,223 @@
+  version: "Dev - Eth2Spec v1.0.1"
+  contact:
+    name: Ethereum Github
+    url: https://github.com/ethereum/eth2.0-apis/issues
+  license:
+    name: "Apache 2.0"
+    url: "https://www.apache.org/licenses/LICENSE-2.0.html"
+
+servers:
+  - url: "{server_url}"
+    variables:
+      server_url:
+        description: "key manager API url"
+        default: "https://public-mainnet-node.ethereum.org"
+
+paths:
+  /eth/v1/keystores:
+    get:
+      operationId: ListKeys
+      summary: List Keys.
+      description: |
+        List all validating pubkeys known to and decrypted by this keymanager binary
+      security:
+        - cookieAuth: []
+      responses:
+        "200":
+          description: Success response
+          content:
+            application/json:
+              schema:
+                title: ListKeysResponse
+                type: object
+                properties:
+                  keys:
+                    type: array
+                    items:
+                      type: object
+                      properties:
+                        validating_pubkey:
+                          $ref: "#/components/types/Pubkey"
+                        derivation_path:
+                          type: string
+                          description: The derivation path (if present in the imported keystore).
+                        not_deletable:
+                          type: boolean
+                          description: The key associated with this pubkey cannot be deleted from the API
+
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "500":
+          $ref: "#/components/responses/InternalError"
+
+    post:
+      operationId: ImportKeystores
+      summary: Import Keystores.
+      description: |
+        Import keystores generated by the Eth2.0 deposit CLI tooling. All keystores MUST be encrypted with the same password.
+
+        Users SHOULD send slashing_protection data associated with the imported pubkeys. MUST follow the format defined in
+        EIP-3076: Slashing Protection Interchange Format.
+      security:
+        - cookieAuth: []
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                keystores:
+                  type: array
+                  description: JSON-encoded keystore files generated with the Launchpad.
+                  items:
+                    type: object
+                passwords:
+                  description: Passwords to unlock imported keystore files. `request.passwords[i]` must unlock `request.keystores[i]`.
+                  items:
+                    type: string
+                slashing_protection:
+                  type: object
+                  required: false
+                  description: Slashing protection of the imported keys data in EIP-3076 JSON format.
```

Comment:
we're going to have to define this here if we want it listed in the schema...

Reply:
I think we could use a [URL reference](https://swagger.io/docs/specification/using-ref/) to https://github.com/eth2-clients/slashing-protection-interchange-tests/blob/master/schema.json